### PR TITLE
HIP-584: HTS GetApprovedPrecompile

### LINKED ISSUE #6581: HIP-584: Provide implementation for GetApprovedPrecompile
- Copy GetApprovedPrecompile in com.hedera.services.
- Integrate GetApprovedPrecompile with the Store interface.
- Add Javadoc describing what was changed in the mirror-node code base
- Create integration tests in ContractCallServicePrecompileTest in hedera-mirror-web3 and validate the returned gas estimates are in the desired range between 5% and 20% more than the actual gas used

### PR DESCRIPTION
**Pull request by mustafauzunn**
**Description**:

Provide implementation for HTS GetApprovedPrecompile

**Related issue(s)**:

Fixes #6581 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/GetApprovedPrecompile.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import com.hedera.services.store.contracts.precompile.codec.GetApprovedResult;
+import com.hedera.services.store.contracts.precompile.codec.RunResult;
+import com.hedera.services.store.contracts.precompile.utils.PrecompilePricingUtils;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import java.math.BigInteger;
+import java.util.Set;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+
+public class GetApprovedPrecompile extends AbstractReadOnlyPrecompile implements EvmGetApprovedPrecompile {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/GetApprovedPrecompile.java:43]: Add javadoc with the differences from the original, especially the need for using the run() method

**Review by IvanKavaldzhiev — DISMISSED** [review]
LGTM, only javadoc should be added

**Review by IvanKavaldzhiev — DISMISSED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]
LGTM, can merge once you fix conflicts and Ivan re-approves. Can almost merge the other two into this one first if you like since they touch all the same files.

**Review by IvanKavaldzhiev — APPROVED** [review]
LGTM

**Pull request merged**